dpnp/backend/backend_iface.hpp

```diff
@@ -813,6 +813,22 @@ INP_DLLEXPORT void custom_rng_laplace_c(void* result, double loc, double scale,
 template <typename _DataType>
 INP_DLLEXPORT void custom_rng_lognormal_c(void* result, _DataType mean, _DataType stddev, size_t size);
 
+/**
+ * @ingroup BACKEND_API
+ * @brief math library implementation of random number generator (multinomial distribution)
+ *
+ * @param [in]  size     Number of elements in `result` arrays.
+ *
+ * @param [in]  ntrial   Number of independent trials.
+ *
+ * @param [in]  p_vector Probability vector of possible outcomes (k length).
+ *
+ * @param [out] result   Output array.
+ *
+ */
+template <typename _DataType>
+INP_DLLEXPORT void custom_rng_multinomial_c(void* result, int ntrial, std::vector<double>& p_vector, size_t size);
```

Comment:
Perhaps it might be fixed later.
This is a not quite good to use STL in the functions interface. These containers are good for regular usage but in our case we need to provide "simple" interface. Usually we use Cython or Numba IR as a customer for this intrface. Importing C++ STL is a quite big pain for such usage model.
Please use `double* p_vector, size_t p_vector_size` instead.